### PR TITLE
Prevent installing on PHP7

### LINKED ISSUE #3451: PHP Fatal Error on app:install
When running ` php artisan app:install` on a freshly installed instance running the newest clone and checked out for `v2.3.9`.

```PHP Fatal error:  Uncaught ErrorException: count(): Parameter must be an array or an object that implements Countable in /var/www/cachet/vendor/graham-campbell/exceptions/src/ExceptionIdentifier.php:51
Stack trace:
#0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'count(): Parame...', '/var/www/cachet...', 51, Array)
#1 /var/www/cachet/vendor/graham-campbell/exceptions/src/ExceptionIdentifier.php(51): count(NULL)
#2 /var/www/cachet/vendor/graham-campbell/exceptions/src/ExceptionHandlerTrait.php(47): GrahamCampbell\Exceptions\ExceptionIdentifier->identify(Object(ErrorException))
#3 /var/www/cachet/bootstrap/cache/compiled.php(2163): GrahamCampbell\Exceptions\ExceptionHandler->report(Object(ErrorException))
#4 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleException(Object(ErrorException))
#5 {main}
  thrown in /var/www/cachet/vendor/graham-campbell/exceptions/src/ExceptionIdentifier.php on line 51
PHP Fatal error:  Uncaught ErrorException: count(): Parameter must be an array or an object that implements Countable in /var/www/cachet/vendor/graham-campbell/exceptions/src/ExceptionIdentifier.php:51
Stack trace:
#0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'count(): Parame...', '/var/www/cachet...', 51, Array)
#1 /var/www/cachet/vendor/graham-campbell/exceptions/src/ExceptionIdentifier.php(51): count(NULL)
#2 /var/www/cachet/vendor/graham-campbell/exceptions/src/ExceptionHandlerTrait.php(47): GrahamCampbell\Exceptions\ExceptionIdentifier->identify(Object(Symfony\Component\Debug\Exception\FatalErrorException))
#3 /var/www/cachet/bootstrap/cache/compiled.php(2163): GrahamCampbell\Exceptions\ExceptionHandler->report(Object(Symfony\Component\Debug\Exception\FatalErrorException))
#4 /var/www/cachet/bootstrap/cache/compiled.php(2181): Illuminate\Foundation\Bootstrap\HandleExceptions->handleException(Object(Symfony\Component\Debug\Exception\FatalErrorException))
#5 [internal function]: I in /var/www/cachet/vendor/graham-campbell/exceptions/src/ExceptionIdentifier.php on line 51
```

System Details:
OS: Ubuntu 18.04
PHP: 
```
php --version
PHP 7.2.10-0ubuntu0.18.04.1 (cli) (built: Sep 13 2018 13:45:02) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.10-0ubuntu0.18.04.1, Copyright (c) 1999-2018, by Zend Technologies
```

PHP INI:
```php --ini
Configuration File (php.ini) Path: /etc/php/7.2/cli
Loaded Configuration File:         /etc/php/7.2/cli/php.ini
Scan for additional .ini files in: /etc/php/7.2/cli/conf.d
Additional .ini files parsed:      /etc/php/7.2/cli/conf.d/10-opcache.ini,
/etc/php/7.2/cli/conf.d/10-pdo.ini,
/etc/php/7.2/cli/conf.d/15-xml.ini,
/etc/php/7.2/cli/conf.d/20-calendar.ini,
/etc/php/7.2/cli/conf.d/20-ctype.ini,
/etc/php/7.2/cli/conf.d/20-dom.ini,
/etc/php/7.2/cli/conf.d/20-exif.ini,
/etc/php/7.2/cli/conf.d/20-fileinfo.ini,
/etc/php/7.2/cli/conf.d/20-ftp.ini,
/etc/php/7.2/cli/conf.d/20-gd.ini,
/etc/php/7.2/cli/conf.d/20-gettext.ini,
/etc/php/7.2/cli/conf.d/20-iconv.ini,
/etc/php/7.2/cli/conf.d/20-json.ini,
/etc/php/7.2/cli/conf.d/20-mbstring.ini,
/etc/php/7.2/cli/conf.d/20-phar.ini,
/etc/php/7.2/cli/conf.d/20-posix.ini,
/etc/php/7.2/cli/conf.d/20-readline.ini,
/etc/php/7.2/cli/conf.d/20-shmop.ini,
/etc/php/7.2/cli/conf.d/20-simplexml.ini,
/etc/php/7.2/cli/conf.d/20-sockets.ini,
/etc/php/7.2/cli/conf.d/20-sysvmsg.ini,
/etc/php/7.2/cli/conf.d/20-sysvsem.ini,
/etc/php/7.2/cli/conf.d/20-sysvshm.ini,
/etc/php/7.2/cli/conf.d/20-tokenizer.ini,
/etc/php/7.2/cli/conf.d/20-wddx.ini,
/etc/php/7.2/cli/conf.d/20-xmlreader.ini,
/etc/php/7.2/cli/conf.d/20-xmlwriter.ini,
/etc/php/7.2/cli/conf.d/20-xsl.ini
```

.env
```
APP_ENV=production
APP_DEBUG=false
APP_URL=<REDACTED>
APP_KEY=<REDACTED>

DB_DRIVER=postgresql
DB_HOST=<REDACTED>
DB_DATABASE=cachet
DB_USERNAME=cachet
DB_PASSWORD=<REDACTED>
DB_PORT=5294
DB_PREFIX=null

CACHE_DRIVER=file
SESSION_DRIVER=file
QUEUE_DRIVER=sync
CACHET_EMOJI=false
CACHET_BEACON=false

MAIL_DRIVER=smtp
MAIL_HOST=localhost
MAIL_PORT=25
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ADDRESS=null
MAIL_NAME=null
MAIL_ENCRYPTION=tls

REDIS_HOST=null
REDIS_DATABASE=null
REDIS_PORT=null

GITHUB_TOKEN=null
```

Steps Already taken:

I have done the following:

> Run the following:
> 1. `composer self-update`
> 2. `composer clear-cache`
> 3. `rm -rf vendor`
> 4. `composer install`

_Originally posted by @jbrooksuk in https://github.com/CachetHQ/Cachet/issues/1577#issuecomment-190617826_

Output:

```
cachet@app-alpha:/var/www/cachet$ composer self-update
You are already using composer version 1.8.3 (stable channel).
cachet@app-alpha:/var/www/cachet$ composer clear-cache
Clearing cache (cache-vcs-dir): /home/cachet/.composer/cache/vcs
Clearing cache (cache-repo-dir): /home/cachet/.composer/cache/repo
Clearing cache (cache-files-dir): /home/cachet/.composer/cache/files
Clearing cache (cache-dir): /home/cachet/.composer/cache
All caches cleared.
cachet@app-alpha:/var/www/cachet$ rm -rf vendor
cachet@app-alpha:/var/www/cachet$ composer install
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Package operations: 97 installs, 0 updates, 0 removals
  - Installing vlucas/phpdotenv (v2.3.0): Downloading (100%)         
  - Installing symfony/polyfill-mbstring (v1.2.0): Downloading (100%)         
  - Installing symfony/var-dumper (v3.0.9): Downloading (100%)         
  - Installing symfony/translation (v3.0.9): Downloading (100%)         
  - Installing symfony/routing (v3.0.9): Downloading (100%)         
  - Installing symfony/process (v3.0.9): Downloading (100%)         
  - Installing symfony/polyfill-util (v1.2.0): Downloading (100%)         
  - Installing symfony/polyfill-php56 (v1.2.0): Downloading (100%)         
  - Installing symfony/http-foundation (v3.0.9): Downloading (100%)         
  - Installing symfony/event-dispatcher (v3.1.3): Downloading (100%)         
  - Installing psr/log (1.0.0): Downloading (100%)         
  - Installing symfony/debug (v3.0.9): Downloading (100%)         
  - Installing symfony/http-kernel (v3.0.9): Downloading (100%)         
  - Installing symfony/finder (v3.0.9): Downloading (100%)         
  - Installing symfony/console (v3.0.9): Downloading (100%)         
  - Installing swiftmailer/swiftmailer (v5.4.3): Downloading (100%)         
  - Installing nikic/php-parser (v2.1.0): Downloading (100%)         
  - Installing jakub-onderka/php-console-color (0.1): Downloading (100%)         
  - Installing jakub-onderka/php-console-highlighter (v0.3.2): Downloading (100%)         
  - Installing dnoegel/php-xdg-base-dir (0.1): Downloading (100%)         
  - Installing psy/psysh (v0.7.2): Downloading (100%)         
  - Installing paragonie/random_compat (v1.4.1): Downloading (100%)         
  - Installing nesbot/carbon (1.21.0): Downloading (100%)         
  - Installing mtdowling/cron-expression (v1.1.0): Downloading (100%)         
  - Installing monolog/monolog (1.21.0): Downloading (100%)         
  - Installing league/flysystem (1.0.26): Downloading (100%)         
  - Installing jeremeamia/superclosure (2.2.0): Downloading (100%)         
  - Installing doctrine/inflector (v1.1.0): Downloading (100%)         
  - Installing classpreloader/classpreloader (3.0.0): Downloading (100%)         
  - Installing laravel/framework (v5.2.39): Downloading (100%)         
  - Installing alt-three/badger (v3.1.0): Downloading (100%)         
  - Installing alt-three/bus (v1.1.0): Downloading (100%)         
  - Installing league/commonmark (0.13.4): Downloading (100%)         
  - Installing psr/http-message (1.0.1): Downloading (100%)         
  - Installing guzzlehttp/psr7 (1.3.1): Downloading (100%)         
  - Installing guzzlehttp/promises (1.2.0): Downloading (100%)         
  - Installing guzzlehttp/guzzle (6.2.1): Downloading (100%)         
  - Installing alt-three/emoji (v3.1.0): Downloading (100%)         
  - Installing alt-three/throttle (v1.0.6): Downloading (100%)         
  - Installing alt-three/validator (v1.5.0): Downloading (100%)         
  - Installing mtdowling/jmespath.php (2.3.0): Downloading (100%)         
  - Installing aws/aws-sdk-php (3.18.37): Downloading (100%)         
  - Installing backup-manager/backup-manager (1.1.1): Downloading (100%)         
  - Installing backup-manager/laravel (1.1.1): Downloading (100%)         
  - Installing barryvdh/laravel-cors (v0.8.1): Downloading (100%)         
  - Installing doctrine/lexer (v1.0.1): Downloading (100%)         
  - Installing doctrine/annotations (v1.2.7): Downloading (100%)         
  - Installing doctrine/cache (v1.6.0): Downloading (100%)         
  - Installing doctrine/collections (v1.3.0): Downloading (100%)         
  - Installing doctrine/common (v2.6.1): Downloading (100%)         
  - Installing doctrine/dbal (v2.5.4): Downloading (100%)         
  - Installing symfony/css-selector (v3.1.3): Downloading (100%)         
  - Installing tijsverkoyen/css-to-inline-styles (1.5.5): Downloading (100%)         
  - Installing fedeisas/laravel-mail-css-inliner (1.5): Downloading (100%)         
  - Installing fideloper/proxy (3.1.0): Downloading (100%)         
  - Installing graham-campbell/security (v3.6.0): Downloading (100%)         
  - Installing graham-campbell/binput (v3.4.0): Downloading (100%)         
  - Installing graham-campbell/core (v5.1.0): Downloading (100%)         
  - Installing graham-campbell/exceptions (v8.8.0): Downloading (100%)         
  - Installing graham-campbell/markdown (v6.1.0): Downloading (100%)         
  - Installing jenssegers/date (v3.2.2): Downloading (100%)         
  - Installing mccool/laravel-auto-presenter (4.3.0): Downloading (100%)         
  - Installing bacon/bacon-qr-code (1.0.1): Downloading (100%)         
  - Installing simplesoftwareio/simple-qrcode (1.3.3): Downloading (100%)         
  - Installing christian-riesen/base32 (1.3.1): Downloading (100%)         
  - Installing pragmarx/google2fa (v0.7.1): Downloading (100%)         
  - Installing twig/twig (v1.24.1): Downloading (100%)         
  - Installing rcrowe/twigbridge (v0.9.2): Downloading (100%)         
  - Installing roumen/feed (v2.10.4): Downloading (100%)         
  - Installing graham-campbell/testbench-core (v1.1.1): Downloading (100%)         
  - Installing alt-three/testbench (v1.8.1): Downloading (100%)         
  - Installing filp/whoops (2.1.3): Downloading (100%)         
  - Installing fzaninotto/faker (v1.6.0): Downloading (100%)         
  - Installing hamcrest/hamcrest-php (v1.2.2): Downloading (100%)         
  - Installing mockery/mockery (0.9.5): Downloading (100%)         
  - Installing webmozart/assert (1.0.2): Downloading (100%)         
  - Installing phpdocumentor/reflection-common (1.0): Downloading (100%)         
  - Installing phpdocumentor/type-resolver (0.2): Downloading (100%)         
  - Installing phpdocumentor/reflection-docblock (3.1.0): Downloading (100%)         
  - Installing phpunit/php-token-stream (1.4.8): Downloading (100%)         
  - Installing symfony/yaml (v3.1.3): Downloading (100%)         
  - Installing sebastian/version (1.0.6): Downloading (100%)         
  - Installing sebastian/global-state (1.1.1): Downloading (100%)         
  - Installing sebastian/recursion-context (1.0.2): Downloading (100%)         
  - Installing sebastian/exporter (1.2.2): Downloading (100%)         
  - Installing sebastian/environment (1.3.7): Downloading (100%)         
  - Installing sebastian/diff (1.4.1): Downloading (100%)         
  - Installing sebastian/comparator (1.2.0): Downloading (100%)         
  - Installing phpunit/php-text-template (1.2.1): Downloading (100%)         
  - Installing doctrine/instantiator (1.0.5): Downloading (100%)         
  - Installing phpunit/phpunit-mock-objects (2.3.8): Downloading (100%)         
  - Installing phpunit/php-timer (1.0.8): Downloading (100%)         
  - Installing phpunit/php-file-iterator (1.4.1): Downloading (100%)         
  - Installing phpunit/php-code-coverage (2.2.4): Downloading (100%)         
  - Installing phpspec/prophecy (v1.6.1): Downloading (100%)         
  - Installing phpunit/phpunit (4.8.21): Downloading (100%)         
  - Installing symfony/dom-crawler (v3.1.3): Downloading (100%)         
symfony/var-dumper suggests installing ext-symfony_debug
symfony/translation suggests installing symfony/config
symfony/routing suggests installing symfony/config (For using the all-in-one router or any loader)
symfony/routing suggests installing symfony/dependency-injection (For loading routes from a service)
symfony/routing suggests installing symfony/expression-language (For using expression matching)
symfony/event-dispatcher suggests installing symfony/dependency-injection
symfony/http-kernel suggests installing symfony/browser-kit
symfony/http-kernel suggests installing symfony/class-loader
symfony/http-kernel suggests installing symfony/config
symfony/http-kernel suggests installing symfony/dependency-injection
psy/psysh suggests installing ext-pdo-sqlite (The doc command requires SQLite to work.)
paragonie/random_compat suggests installing ext-libsodium (Provides a modern crypto API that can be used to generate random bytes.)
monolog/monolog suggests installing doctrine/couchdb (Allow sending log messages to a CouchDB server)
monolog/monolog suggests installing ext-amqp (Allow sending log messages to an AMQP server (1.0+ required))
monolog/monolog suggests installing ext-mongo (Allow sending log messages to a MongoDB server)
monolog/monolog suggests installing graylog2/gelf-php (Allow sending log messages to a GrayLog2 server)
monolog/monolog suggests installing mongodb/mongodb (Allow sending log messages to a MongoDB server via PHP Driver)
monolog/monolog suggests installing php-amqplib/php-amqplib (Allow sending log messages to an AMQP server using php-amqplib)
monolog/monolog suggests installing php-console/php-console (Allow sending log messages to Google Chrome)
monolog/monolog suggests installing rollbar/rollbar (Allow sending log messages to Rollbar)
monolog/monolog suggests installing ruflin/elastica (Allow sending log messages to an Elastic Search server)
monolog/monolog suggests installing sentry/sentry (Allow sending log messages to a Sentry server)
league/flysystem suggests installing league/flysystem-aws-s3-v2 (Allows you to use S3 storage with AWS SDK v2)
league/flysystem suggests installing league/flysystem-aws-s3-v3 (Allows you to use S3 storage with AWS SDK v3)
league/flysystem suggests installing league/flysystem-azure (Allows you to use Windows Azure Blob storage)
league/flysystem suggests installing league/flysystem-cached-adapter (Flysystem adapter decorator for metadata caching)
league/flysystem suggests installing league/flysystem-copy (Allows you to use Copy.com storage)
league/flysystem suggests installing league/flysystem-dropbox (Allows you to use Dropbox storage)
league/flysystem suggests installing league/flysystem-eventable-filesystem (Allows you to use EventableFilesystem)
league/flysystem suggests installing league/flysystem-rackspace (Allows you to use Rackspace Cloud Files)
league/flysystem suggests installing league/flysystem-sftp (Allows you to use SFTP server storage via phpseclib)
league/flysystem suggests installing league/flysystem-webdav (Allows you to use WebDAV storage)
league/flysystem suggests installing league/flysystem-ziparchive (Allows you to use ZipArchive adapter)
laravel/framework suggests installing league/flysystem-aws-s3-v3 (Required to use the Flysystem S3 driver (~1.0).)
laravel/framework suggests installing league/flysystem-rackspace (Required to use the Flysystem Rackspace driver (~1.0).)
laravel/framework suggests installing pda/pheanstalk (Required to use the beanstalk queue driver (~3.0).)
laravel/framework suggests installing predis/predis (Required to use the redis cache and queue drivers (~1.0).)
laravel/framework suggests installing pusher/pusher-php-server (Required to use the Pusher broadcast driver (~2.0).)
laravel/framework suggests installing symfony/psr-http-message-bridge (Required to psr7 bridging features (0.2.*).)
league/commonmark suggests installing league/commonmark-extras (Library of useful extensions including smart punctuation)
aws/aws-sdk-php suggests installing aws/aws-php-sns-message-validator (To validate incoming SNS notifications)
aws/aws-sdk-php suggests installing ext-curl (To send requests using cURL)
backup-manager/backup-manager suggests installing league/flysystem-aws-s3-v3 (AwsS3 and GoogleCS adapter support.)
backup-manager/backup-manager suggests installing league/flysystem-dropbox (Dropbox adapter support.)
backup-manager/backup-manager suggests installing league/flysystem-rackspace (Rackspace adapter support.)
backup-manager/backup-manager suggests installing league/flysystem-sftp (Sftp adapter support.)
rcrowe/twigbridge suggests installing laravelcollective/html (For bringing back html/form in Laravel 5.x)
rcrowe/twigbridge suggests installing twig/extensions (~1.0)
filp/whoops suggests installing whoops/soap (Formats errors as SOAP responses)
sebastian/global-state suggests installing ext-uopz (*)
phpunit/phpunit-mock-objects suggests installing ext-soap (*)
phpunit/php-code-coverage suggests installing ext-xdebug (>=2.2.1)
phpunit/phpunit suggests installing phpunit/php-invoker (~1.1)
Generating autoload files
> Illuminate\Foundation\ComposerScripts::postInstall
> php artisan optimize --force
Generating optimized class loader
Compiling common classes
> php artisan config:cache
Configuration cache cleared!
Configuration cached successfully!
> php artisan route:cache
Route cache cleared!
Routes cached successfully!
```

### PR DESCRIPTION
**Pull request by jbrooksuk**
Closes #3451

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -93,9 +93,6 @@
         ]
     },
     "config": {
-        "platform": {
```
Comment (GrahamCampbell) [inline composer.json]: Don't revert this. This is needed.

File: composer.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
         }
     ],
     "require": {
-        "php": ">=5.5.9",
+        "php": ">=5.5.9 <7.0",
```
Comment (GrahamCampbell) [inline composer.json]: I think Cachet works on 7.0 but not 7.1+. I can't remember which version had the `count` changes.

Comment (jbrooksuk) [reply composer.json]: 7.2 I think?

File: composer.json (hunk 1 above)
Comment (jbrooksuk) [reply composer.json]: Is this not forcing 5.5.9 though?

Comment (GrahamCampbell) [reply composer.json]: Yes, it forces the dependencies to be resolved as if you were 5.5.9. Without this, the lock file will contain dependencies that don't work on PHP 5.5.

Comment (jbrooksuk) [reply composer.json]: But does that not conflict with the requirements of PHP 7?

Comment (GrahamCampbell) [reply composer.json]: No?

**Comment by jbrooksuk** [comment]
@GrahamCampbell are the changes okay now?

**Comment by GrahamCampbell** [comment]
Does this actually prevent installation on PHP 7? Have you tried it. I suspect the platform config that is necessary to ensure the dependencies work on PHP 5.5 has the side effect of allowing composer to install even when the version of PHP is incompatible.

**Comment by GrahamCampbell** [comment]
If this is the case, we could just bake compatibility tests and warnings on the setup cli command and web page?

**Comment by GrahamCampbell** [comment]
Cachet works fine on PHP 7.1.x. StyleCI's status page is currently running the 2.3 branch.

**Comment by GrahamCampbell** [comment]
It looks like the original issue was using a way old version of 2.3, from 2016...

**Comment by jbrooksuk** [comment]
Thanks for clarifying @GrahamCampbell!

**Pull request closed without merging**